go/externals/proof_support_generic.go

```diff
@@ -0,0 +1,97 @@
+package externals
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	libkb "github.com/keybase/client/go/libkb"
+	keybase1 "github.com/keybase/client/go/protocol/keybase1"
+	jsonw "github.com/keybase/go-jsonw"
+)
+
+//=============================================================================
+// GenericProof
+//
+
+type GenericChecker struct {
+	proof libkb.RemoteProofChainLink
+}
+
+var _ libkb.ProofChecker = (*GenericChecker)(nil)
+
+func NewGenericChecker(p libkb.RemoteProofChainLink) (*GenericChecker, libkb.ProofError) {
+	return &GenericChecker{p}, nil
+}
+
+func (rc *GenericChecker) GetTorError() libkb.ProofError { return nil }
+
+func (rc *GenericChecker) CheckStatus(m libkb.MetaContext, h libkb.SigHint, _ libkb.ProofCheckerMode, pvlU libkb.PvlUnparsed) libkb.ProofError {
+	// TODO will have to use the `check_url`/`check_path` config values to verify the proof.
+	return nil
+}
+
+//
+//=============================================================================
+
+type GenericServiceType struct {
+	libkb.BaseServiceType
+	conf       keybase1.GenericServiceConfig
+	usernameRe *regexp.Regexp
+}
+
+func NewGenericServiceType(conf keybase1.GenericServiceConfig) GenericServiceType {
+	return GenericServiceType{
+		conf:       conf,
+		usernameRe: regexp.MustCompile(conf.UsernameRe),
```

Comment:
will this be a pain to share between iced/go?

Reply:
Oh I was looking for but missed this.. I don't think this should use `MustCompile` since it makes it too easy for the server to crash all the clients. The client may want to do other validation against the server's response. Perhaps a validation function that, if it fails, just removes the service from the list would be good.

Reply:
I think that versioning and compatibility will need to be addressed so in that way it will be an unavoidable pain for both sides to have a representation.

Reply:
makes sense, i'll leave as is and drop a todo when we implement the loading